lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2488,8 +2488,9 @@ public function get_root_layout_rules( $selector, $block_metadata ) {
 		$has_block_gap_support = _wp_array_get( $this->theme_json, array( 'settings', 'spacing', 'blockGap' ) ) !== null;
 		if ( $has_block_gap_support ) {
 			$block_gap_value = static::get_property_value( $this->theme_json, array( 'styles', 'spacing', 'blockGap' ) );
-			$css            .= '.wp-site-blocks > * { margin-block-start: 0; margin-block-end: 0; }';
-			$css            .= ".wp-site-blocks > * + * { margin-block-start: $block_gap_value; }";
+			$css            .= ":where(.wp-site-blocks) > * { margin-block-start: $block_gap_value; }";
```

Comment:
It looks like this and the JS version are missing the `margin-block-end: 0;` rule, so the user agent stylesheet rules for paragraph are poking through:

<img width="905" alt="image" src="https://user-images.githubusercontent.com/14988353/229409318-ad87677f-2ba6-4d2e-9dcd-bca099b951aa.png">

Reply:
Oh whoops, I copied the old rule and forgot to add the end margin 😳

Reply:
Fixed!